src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Binders/DictionaryExpansionConfigurationProviderTests.cs

```diff
@@ -136,6 +136,33 @@ public void GivenAMockConfigurationProvider_WhenInitialized_ThenEnsureJustVariab
             Assert.Equal(variablesInJsonFormat, numberOfJsonVariablesInMockProvider);
         }
 
+        [Theory]
+        [InlineData("")]
+        [InlineData("{}")]
+        [InlineData("{ {")]
+        [InlineData("{ { } }")]
+        [InlineData(" {} ")]
+        [InlineData(" { } ")]
+        [InlineData("{foo}")]
+        [InlineData(" { foo } ")]
+        [InlineData("{ 'foo' }")]
+        [InlineData("{ \"foo\" }")]
+        [InlineData("{070741cc-f9e9-516d-9f32-f3a4cca9614d}")]
+        public void GivenAnInvalidJsonConfiguration_WhenInitialized_ThenKeepValuesAsTheyAre(string value)
+        {
+            MockConfigurationProvider mockProvider = new MockConfigurationProvider(
+                new Dictionary<string, string>()
+                {
+                    { "key", value},
+                });
+
+            DictionaryExpansionConfigurationProvider configurationProvider = new DictionaryExpansionConfigurationProvider(mockProvider);
+
+            configurationProvider.Load();
+
+            Assert.Empty(configurationProvider.GetChildKeys(Array.Empty<string>(), null));
+        }
+
```

Comment:
For some reason I can't add this comment where needed on line 72. :-(
function HasJsonStructure is still used on line 72 and fails on my machine for the original discovery of this issue. Seems like we should remove the function entirely at this point.

Reply:
Fixed the issue adding a call to the new tryparse method.

Reply:
👍